scripts/generate-locales.ts

```diff
@@ -0,0 +1,118 @@
+import ts from 'typescript';
+import path, { relative } from 'path';
+import { writeFile, mkdir, stat } from 'fs/promises';
+import { glob, chalk } from 'zx';
+
+const SRC_DIR = path.resolve('./src/lib/i18n/locales');
+const DEST_DIR = path.resolve('./static/i18n/locales');
+
+const logAndExit = (msg: string, logError: (msg: string) => void) => {
```

Comment:
We copied this from the Temporal CLI download script right? I wonder if we should break this out and put it… somewhere? (Not for this PR, of course.)

Reply:
No, it's slightly different because this script can be run via `pnpm` or as a vite plugin during the build/serve commands. Vite plugins' Context have an `error` function: https://rollupjs.org/plugin-development/#this-error so I'm passing it through to here instead of using `console.error` when running as a vite plugin. This way any vite cleanup stuff can still happen normally instead of immediately exiting the process.